osde2e/managed_cluster_validating_webhooks_test.go

```diff
@@ -201,20 +217,17 @@ var _ = Describe("Managed Cluster Validating Webhooks", Ordered, func() {
 			}
 		})
 
-		withNamespace := func(pod *v1.Pod, namespace string) *v1.Pod {
-			pod.SetNamespace(namespace)
-			return pod
-		}
-
 		It("blocks pods scheduled onto master/infra nodes", func(ctx context.Context) {
-			err := dedicatedAdmink8s.Create(ctx, withNamespace(pod, privilegedNamespace))
+			err := dedicatedAdmink8s.Create(ctx, pod)
 			Expect(errors.IsForbidden(err)).To(BeTrue())
 
-			err = userk8s.Create(ctx, withNamespace(pod, privilegedNamespace))
+			err = userk8s.Create(ctx, pod)
 			Expect(errors.IsForbidden(err)).To(BeTrue())
 
-			err = userk8s.Create(ctx, withNamespace(pod, unprivilegedNamespace))
+			pod.SetNamespace(unprivilegedNamespace)
+			err = userk8s.Create(ctx, pod)
 			Expect(errors.IsForbidden(err)).To(BeTrue())
+			pod.SetNamespace(privilegedNamespace)
```

Comment:
This is the only test that doesn't use the privilegedNamespace, so I'm moved the responsibility of toggling the ns to it.